pkg/apis/monitoring/v1/types.go

```diff
@@ -1015,3 +1015,55 @@ const (
 	SelectorMechanismRelabel SelectorMechanism = "RelabelConfig"
 	SelectorMechanismRole    SelectorMechanism = "RoleSelector"
 )
+
+// WorkloadResourceReference is a link of workload resource wuth configuration resource.
+// +k8s:openapi-gen=true
+type WorkloadResourceReference struct {
+	// The group of the referenced resource.
+	// +kubebuilder:validation:Enum=monitoring.coreos.com
+	// +required
+	Group string `json:"group"`
+	// The type of resource being referenced (e.g. Prometheus or PrometheusAgent).
+	// +kubebuilder:validation:Enum=prometheuses;prometheusagents
```

Comment:
(nit) If we are planning on each `WorkloadResourceReference` referencing one resource, it might be a bit confusing to see plural forms here. 
```suggestion
	// +kubebuilder:validation:Enum=prometheus;prometheusagent
```